go/kbfs/libkbfs/disk_block_cache_wrapped.go

```diff
@@ -322,14 +326,17 @@ func (cache *diskBlockCacheWrapped) Delete(ctx context.Context,
 	// caches. So we use a read lock.
 	cache.mtx.RLock()
 	defer cache.mtx.RUnlock()
-	if cacheType == DiskBlockAnyCache || cacheType == DiskBlockSyncCache {
+	if cache.syncCache != nil &&
+		(cacheType == DiskBlockAnyCache || cacheType == DiskBlockSyncCache) {
 		numRemoved, sizeRemoved, err = cache.syncCache.Delete(ctx, blockIDs)
 		if err != nil {
 			return 0, 0, err
 		}
 		if cacheType == DiskBlockSyncCache {
 			return numRemoved, sizeRemoved, err
 		}
+	} else if cacheType == DiskBlockSyncCache {
+		return 0, 0, errors.New("Sync cache not enabled")
```

Comment:
I'd prefer for this to be the first case, since it requires less mental state tracking as you read the code:
```go
if cacheType == DiskBlockSyncCache && cache.syncCache == nil {
    return 0, 0, errors.New("Sync cache not enabled")
} else if cacheType == DiskBlockAnyCache || cacheType == DiskBlockSyncCache {
```

Reply:
Sure.  The second case still needs the `cache.syncCache != nil` clause though, for the `DiskBlockAnyCache` case, otherwise it could crash.

Reply:
Ah yeah, even better then since it includes all its conditionals.